osf/external/chronos.py

```diff
@@ -0,0 +1,299 @@
+from __future__ import unicode_literals
+import requests
+
+from django.db import transaction
+
+from osf.models import ChronosJournal
+from osf.models import ChronosSubmission
+from osf.utils.workflows import ChronosSubmissionStatus, ReviewStates
+from addons.osfstorage.models import OsfStorageFile
+from website.settings import (
+    DOMAIN, CHRONOS_USE_FAKE_FILE, CHRONOS_FAKE_FILE_URL,
+    CHRONOS_API_KEY, CHRONOS_USERNAME, CHRONOS_PASSWORD, CHRONOS_HOST, VERIFY_CHRONOS_SSL_CERT
+)
+
+
+class ChronosSerializer(object):
+
+    @classmethod
+    def serialize_manuscript(cls, journal_id, preprint, status=ChronosSubmissionStatus.DRAFT.value):
+        """Serialize an OSF preprint for submission to Chronos
+
+        It is currently unclear what ARTICLE_TYPE should be:
+        Possible options are:
+        * abstract
+        * addendum
+        * analytic-perspective
+        * announcement
+        * article-commentary
+        * Book
+        * brief-report
+        * case-report
+        * collection
+        * correction
+        * discussion
+        * dissertation
+        * editorial
+        * in-brief
+        * introduction
+        * letter
+        * meeting-report
+        * news
+        * oration
+        * partial-retraction
+        * product-review
+        * rapid-communication
+        * reply
+        * reprint
+        * research-article
+        * retraction
+        * review-article
+        * study-protocol
+
+        Returns:
+            dict: The serialized manuscript
+        """
+        return {
+            'AUTHORS': [
+                cls.serialize_author(contrib)
+                for contrib in preprint.contributor_set.filter(visible=True).select_related('user')
+            ],
+            'MANUSCRIPT_FILES': [
+                cls.serialize_file(preprint, file_node)
+                for file_node in OsfStorageFile.objects.filter(preprint=preprint)
```

Comment:
Umm... I believe it should just send the primary file.

Reply:
Also, `OsfStorageFile.objects.filter(preprint=preprint)` would not catch files on the supplemental node that is stored on other providers, right?

Reply:
I don't understand the question.  This query should be picking up files that are linked to the preprint, not on the node.  But to send in the primary file, I would just do `preprint.primary_file`, because this might pick up other files stored on the preprint.

Reply:
@pattisdr Talked to Sara about this. We want to submit the preprint primary file and the files on supplemental nodes to Chronos and skip addon files. Does `OsfStorageFile.objects.filter(preprint=preprint)` take care of that, or more checks/queries are needed?

Reply:
no, that's just files on the preprint, gimme a minute..

Reply:
I see what you're saying. So we should serialize only `preprint.primary_file` and use a query to get `OsfStorageFile` on a supplemental node and then serialize them.

Reply:
something like this for the node files:
```python
node_ctype = ContentType.objects.get_for_model(AbstractNode)
OsfStorageFile.objects.filter(target_object_id=preprint.node.id, target_content_type=node_ctype, deleted_on__isnull=True)
```
*however*, the user submitting the preprint may not have permission to view the node -  you can have a preprint with supplemental files, but I'm the only admin on the supplemental node and I've made it private.  So we'd need to check that the submitting user has permissions to the supplemental node (what perms should they have - admin?)

Also, I would push back against sending all supplemental nodes to Chronos. If this is something Product is unsure about if they want it, the supplemental node could have a huge file tree on it, could be problematic.

Reply:
I'll check with prod again with the concerns.

Reply:
Checked with product. Submit only the preprint primary file is fine, as Chronos and APA have yet to define what kind of supplemental data they want on a submission.

Reply:
great. way simpler.

Reply:
I seem to recall discussing this with Chris before he left, I think it was an intentional decision to send all files attached to a preprint.

I can't say whether or not it was the correct decision, though.

Reply:
I'm confused about "all files attached to the preprint", because when Chris wrote it, this was pre-NPD, so it was all files attached to the node, which  include the preprint file, since it was stored on the node as well. Nothing was stored on the preprint.
```
  'MANUSCRIPT_FILES': [
                cls.serialize_file(preprint, file_node)
                for file_node in OsfStorageFile.objects.filter(node=preprint.node)
            ],
```

Post-NPD, the user may not even have permissions to the files on the node, now that preprint perms and the supplemental node perms are separate.